### PR TITLE
Quentin - DB classes

### PR DESCRIPTION
**Pull request by QuentCov**

### CODE REVIEW

File: src/data/CreditCardsDB.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+			    return createCard(rs);
+			}
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+		return null;
+	}
+	
+	public static CreditCard getCreditCard(CreditCard creditCard) {
+		return getCreditCard(creditCard.getCardNumber());
```
Comment (DavidCao22) [inline src/data/CreditCardsDB.java]: Not sure what this does since it gets a CreditCard and returns a CreditCard.

File: src/data/MovieDB.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+			    createMovie(rs);
+			}
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+		return null;
+	}
+	
+	public static Movie getUser(Movie movie) {
+		return getMovie(movie.getName());
```
Comment (DavidCao22) [inline src/data/MovieDB.java]: Shouldn't this return the User? Also, there's no User property on a Movie

File: src/data/MovieDB.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		return movie;
+	}
+	
+	public static Movie getMovie(int id) {
+		String query = "SELECT * FROM movies WHERE Id=" + id;
+		ResultSet rs = Database.runQuery(query);
+		try {
+			if(rs.next())
+			{
+				createMovie(rs);
```
Comment (DavidCao22) [inline src/data/MovieDB.java]: Shouldn't this return the Movie object?

File: src/data/MovieShowingDB.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		return showing;
+	}
+	
+	public static ArrayList<MovieShowing> getMovieShowings(int showroomId) {
+		String query = "SELECT * FROM movieShowings WHERE showroomId=" + showroomId;
+		ResultSet rs = Database.runQuery(query);
+		try {
+			while(rs.next())
+			{
+			    createMovieShowing(rs);
```
Comment (DavidCao22) [inline src/data/MovieShowingDB.java]: needs to return the ArrayList

File: src/data/OrdersDB.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+			    return order;
+			}
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+		return null;
+	}
+	
+	public static Order getOrder(Order order) {
+		return getOrder(order.getCustomer().getEmailAddress());
```
Comment (DavidCao22) [inline src/data/OrdersDB.java]: Isn't this also getting an Order as a param and returning the same Order?

File: src/data/TheatresDB.java [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+						    + "theatreBuildings.City, theatreBuildings.State, theatreBuildings.ZipCode,"
+						    + "theatreBuildings.Name, showrooms.Id, showrooms.Name, showrooms.AvailableSeats,"
+						    + "FROM theatreBuildings t WHERE Name=" + name 
+						    + "INNER JOIN showrooms ON theatreBuildings.Id=showrooms.theatreId";
+		ResultSet rs = Database.runQuery(query);
+		return createTheatre(rs);
+	}
+	
+	public static Theatre getTheatre(Theatre theatre) {
+		return getTheatre(theatre.getName());
```
Comment (DavidCao22) [inline src/data/TheatresDB.java]: Isn't this getting a Theatre object as a param and returning the same one?

File: src/data/UserDB.java [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+		return null;
+	}
+	
+	public static User getUser(int id) {
+		String query = "SELECT * FROM users WHERE Id=" + id;
+		ResultSet rs = Database.runQuery(query);
+		try {
+			if(rs.next())
+			{
+			    createUser(rs);
```
Comment (DavidCao22) [inline src/data/UserDB.java]: Needs to return the User

File: src/data/UserDB.java [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+		return null;
+	}
+	
+	public static User getUser(String userName) {
+		String query = "SELECT * FROM users WHERE EmailAddress=" + userName;
+		ResultSet rs = Database.runQuery(query);
+		try {
+			if(rs.next())
+			{
+			    createUser(rs);
```
Comment (DavidCao22) [inline src/data/UserDB.java]: Needs to return the User

File: src/data/UserDB.java [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+			}
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+		return null;
+	}
+	
+	public static User getUser(User user) {
+		return getUser(user.getEmailAddress());
```
Comment (DavidCao22) [inline src/data/UserDB.java]: Isn't this getting a User object as a param and returning the same object?

File: src/models/User.java [hunk 10]
```diff
@@ -14,6 +8,10 @@
 	private Address streetAddress;
 	private String phoneNumber;
 
+	public User() {
+		super();
```
Comment (DavidCao22) [inline src/models/User.java:12]: Do we need to add this?

**Review by DavidCao22 — CHANGES_REQUESTED** [review]

File: src/models/User.java (hunk 10 above)
Comment (QuentCov) [reply src/models/User.java:12]: JavaBeans need an empty constructor.

**Comment by QuentCov** [comment]
That should be all the changes

**Review by DavidCao22 — APPROVED** [review]

**Comment by DavidCao22** [comment]
Looks good to me.

**Pull request merged**